src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -1305,16 +1298,15 @@ struct TopologyMetadata
             dim_conn.set(DataType(int_dtype.id(), dim_buffers[di].size()));
             data_conn.to_data_type(int_dtype.id(), dim_conn);
 
-            // Generate size for polygonal mesh
-            if (di == 2)
+            // Initialize element/sizes for polygonal mesh using polyhedral's
+            // subelement/sizes
+            if (di == 2 && topo_shape.is_polyhedral())
```

Comment:
Just to clarify @bmhan12, the code within this conditional basically just hooks up the `dim_topos` polygonal `Node` such that it points to the `dim_topos` polyhedral subelement `Node`, which allows the `offsets` generated to properly propagate in all places that it's needed (i.e. the polyhedral 3D "subelements" entry in `dim_topos` and the polygonal 2D "elements" entry in `dim_topos`), correct? Just want to make sure I understand; thanks for looking into this so quickly!

to hook up the correct path in the 3D case so that offsets are generated for the

Reply:
Yes!

Reply:
Excellent, thanks Brian!